### ISSUE TITLE
Look into BigQuery "authorized views"

### ISSUE THREAD

**Issue by nevillelyh** [spotify/scio#1226]
Which might solve permission issues when compiling code that uses BigQuery macros.
https://cloud.google.com/bigquery/docs/share-access-views

**Comment by francescoperera** [comment]
@nevillelyh  I am interested in this topic. Can you give more context in what you are looking to achieve  here ? i.e add ability to create an authorized view from the client, or a method on the Scollection to save output to a view ( which is really just saving it to a normal table and exposing that table as a view)

**Comment by nevillelyh** [comment]
This is more of an internal use case i.e. granting limited views to sensitive data sets without full access. Most of the work will probably be done in our internal data catalog system. Not sure how much work is really needed in Scio.

**Comment by nevillelyh** [comment]
@francescoperera Also curious if your company has similar problem with ACL management and if something like authorized views can help.

**Comment by francescoperera** [comment]
@nevillelyh  yes we use overlay authorized views on top of any table/source data that we ingest. Data consumers only have access to authorized views. Also the views are stored in a completely different project from the source data.
It helps us with GDPR concerns and other sensitive information as the we can hide columns or put dummy values to columns that are too sensitive. End-users limited by what the sql query in the view gives them ( unless they have access to the source table)
What we have done is exposed both a limited view  and full view of a sensitive table ( stored in different datasets).
i.e lets say we need to expose our profile table.
  - in `processed_dataset` we store the limited view ( where we hide columns like first name, last name)
 - in `secure_dataset` we store the full view ( no columns are hidden)

Anyone in the company can access the `processed_dataset` but not everyone has access to secure. If your work requires a full dataset, then you get added to special google-group ( this goes through an approval/vetting process) which is already added to the secure_dataset
